test/telemetry/src/main/java/org/opensearch/test/telemetry/tracing/StrictCheckSpanProcessor.java

```diff
@@ -28,7 +28,10 @@ public StrictCheckSpanProcessor() {}
 
     @Override
     public void onStart(Span span) {
-        spanMap.put(span.getSpanId(), toMockSpanData(span));
+        // TODO: We should be able to address this with Sampler.
```

Comment:
What is that for?

Reply:
Actually, StricktCheckSpanProcessor check if all the spans opened by a particular test cases ended or not. I have seen that some internal transport actions are creating spans and closing sometimes later that this check and eventually some tests are failing intermittently. So removed the internal transport action spans from validation for now.

Reply:
This is concerning, internal or not - every reported span must be closed, please remove this filtering and let us understand the deterministic point to run this check.

Reply:
Agreed, need to spend time on this.

Reply:
Shall I fix it with Transport instrumentation PR as that is half baked but requires this PR to be merged first.

Reply:
@Gaganjuneja we cannot commit such changes, let us split the changeset if you could not fix it now

Reply:
Let me spend some time on that.

Reply:
@reta, There are three ways we can solve this. Your thoughts please.

1. We should introduce the shutdown behaviour in MockTracingTelemetry and SpanProcessor which would stop taking the new SpanRequests and will wait for 10 seconds to clear up the existing opened and buffered spans. Otel does this way. I have added code changes for this and please go through it once. [10 seconds is the decent period used everywhere in the OTEL and OpenSearch ]
2. We should close the Tracer inside a awaitClose method in Node.java https://github.com/opensearch-project/OpenSearch/blob/c90b6eac285055959d419ae67c369421355e78f8/server/src/main/java/org/opensearch/node/Node.java#L1534 once the lifecycle is closed. [Little out of the way.]
3. We should expose the tracing telemetry instance out of Node.java to its implementers and then close this in the teardown method in OpenSearchSingleNodeTestCase.java and InternalTestCluster.java where the MockNode is being created. [Not very clean]

Reply:
Thanks @Gaganjuneja , 

> 1) We should introduce the shutdown behaviour in MockTracingTelemetry and SpanProcessor which would stop taking the new SpanRequests and will wait for 10 seconds to clear up the existing opened and buffered spans

Definitely make sense, but please implement it over `Closeable`  (commented on that)